clsp/fusion_singleton.clsp

```diff
@@ -31,149 +26,98 @@
   (include condition_codes.clib)
   (include utils.clib)
 
-  (defun-inline base_conditions (singleton_coin_id SINGLETON_MOD_HASH LAUNCHER_ID LAUNCHER_PUZZLE_HASH singleton_inner_puzzlehash) 
+  (defun-inline base_conditions (singleton_coin_id SINGLETON_STRUCT singleton_inner_puzzlehash)
     (list
       ; block ephemeral spends
       (list ASSERT_HEIGHT_RELATIVE 1)
 
       ; who am I?
       (list ASSERT_MY_COIN_ID singleton_coin_id)
-      (list ASSERT_MY_PUZZLEHASH (calculate_full_puzzle_hash SINGLETON_MOD_HASH LAUNCHER_ID LAUNCHER_PUZZLE_HASH singleton_inner_puzzlehash))
-      (list ASSERT_MY_AMOUNT 1)
+      (list ASSERT_MY_PUZZLEHASH (calculate_full_puzzle_hash SINGLETON_STRUCT singleton_inner_puzzlehash))
+      (list ASSERT_MY_AMOUNT 1) ; This feels unnecessary. And if you ever create such a singleton with amount > 1 it is locked forever.
 
       ; spend myself forward
-      (list CREATE_COIN singleton_inner_puzzlehash 1 (list LAUNCHER_ID)) ;; opt to keep details beyond launcher ID encoded in eve coin spend
+      (list CREATE_COIN singleton_inner_puzzlehash 1 (list (f (r SINGLETON_STRUCT)))) ;; opt to keep details beyond launcher ID encoded in eve coin spend
```

Comment:
is this needed? can always be found in the parent's puzzle reveal

Reply:
The puzzle doesn't need it. @trgarrett, are you planning to use the hint for easier lookups?